.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,13 @@
+name: CI to build the sdk
```

Comment:
The pull request should also run the build step. Only the publish job has a condition for only tags. Here is the documentation for circleci about it https://circleci.com/docs/workflows/#executing-workflows-for-a-git-tag 

```
test:
    jobs:
      - setup:
          filters:
            tags:
              only: /v.*/
      - build:
          requires:
            - setup
          filters:
            tags:
              only: /v.*/
      - publish:
          context:
            - gcp-prod
            - npm-prod
          requires:
            - build
          filters:
            branches:
              ignore: /.*/
            tags:
              only: /v.*/
              ```